fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVService.java

```diff
@@ -207,12 +209,19 @@ public void processEvent(Event event) throws EventException {
         }
     }
 
-    private boolean canRefresh(MTMV mtmv, TableIf table)   {
+    private boolean canRefresh(MTMV mtmv, TableIf table) {
         if (mtmv.getExcludedTriggerTables().contains(table.getName())) {
             LOG.info("skip refresh mtmv: {}, because exclude trigger table: {}",
                     mtmv.getName(), table.getName());
             return false;
         }
+        // replace/alter base table,not change MTMVRelation, only change MTMV to schema_change,
+        // Therefore, it may trigger incorrect materialized view refresh
+        if (mtmv.getStatus().getState().equals(MTMVState.SCHEMA_CHANGE)) {
+            LOG.info("skip refresh mtmv: {}, because state is SCHEMA_CHANGE, trigger table: {}",
+                    mtmv.getName(), table.getName());
+            return false;
+        }
```

Comment:
if we do not refresh when status is `SCHEMA_CHANGE`, how can we refresh mtmv that status be set to `SCHEMA_CHANGE` when upgrade?

Reply:
only handle commit trigger